From e499778e32e35654fec0ce79ff734b52a7a5206d Mon Sep 17 00:00:00 2001
From: ariel aviad <aaviad@habana.ai>
Date: Mon, 02 Sep 2024 10:48:15 +0300
Subject: [PATCH] [FSW-4085] gaudi2: rom sha: define sha_sum_len_rom

Prepare to remove sha256 we need to remove
all uses of sha256
replace using SHA256_SUM_LEN and create new define
SHA_SUM_LEN_ROM for reading sec key hash in ROM

the reason for the special define is that
if security keys hash location is ROM then only the first
256 bits are compared against the hash calculated.
Since there are only 256 bits available in PSOC registers
to store keys hash

Change-Id: Ie04bde3d6b5a5a2cba1b327eb9d9beacad3ff1e3
Signed-off-by: ariel aviad <aaviad@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index b1cc8ce..02fd130 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -21,7 +21,6 @@
 #include <efuse.h>
 #include <psoc_security_regs.h>
 #include "sha.h"
-#include <sha256.h>
 #include <gaudi2/gaudi2_security.h>
 #include <preboot/fit.h>
 #include <preboot/fit_rsa.h>
@@ -645,7 +644,7 @@
 	 * available in PSOC registers to store keys hash
 	 */
 	if (otp->num_of_keys == SEC_KEY_HASH_COUNT_IN_ROM)
-		hash_compare_len = SHA256_SUM_LEN;
+		hash_compare_len = SHA_SUM_LEN_ROM;
 
 	return !memcmp(&otp->keys[key_idx * SHA_SUM_LEN], value, hash_compare_len);
 }
diff --git a/arch/gaudi2/common/sha.h b/arch/gaudi2/common/sha.h
index 301a054..eb392e1 100644
--- a/arch/gaudi2/common/sha.h
+++ b/arch/gaudi2/common/sha.h
@@ -6,6 +6,8 @@
 #include <sha512.h>
 #include <preboot/fit_rsa.h>
 
+/* at ROM there are only 256 bits available in PSOC registers for keys hash */
+#define SHA_SUM_LEN_ROM			(32)
 #define SHA_SUM_LEN			SHA384_SUM_LEN
 #define SHA_DER_LEN			SHA384_DER_LEN
 #define SHA_DER_PRE			sha384_der_prefix
